lib/still/preprocessor/profiler.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Still.Preprocessor.Profiler do
+  alias Still.Preprocessor
+  alias Still.Profiler
+  alias Still.SourceFile
+
+  use Preprocessor
+
+  @impl true
+  def render(%{metadata: metadata} = file) do
+    if should_profile?(file) do
+      start_time = Profiler.timestamp()
+
+      metadata = Map.put(metadata, :start_time, start_time)
+
+      %{file | metadata: metadata}
+    else
+      file
+    end
+  end
+
+  @impl true
+  def after_render(%{metadata: %{start_time: start_time}} = file) do
+    if should_profile?(file) do
```

Comment:
Isn't this redundant? If there's a `start_time` variable, this check already ran.

Reply:
true 👍